jvm/ml4ir-inference/src/test/integration/ml4ir/inference/tensorflow/TensorFlowInferenceIT.scala

```diff
@@ -53,11 +89,109 @@ class TensorFlowInferenceIT extends TestData {
+    val featureConfigPath = generatedBundleLocation + "ml4ir/applications/ranking/tests/data/configs/feature_config.yaml"
+
+    val allScores: Iterable[(StringMapQueryAndPredictions, SequenceExample, Array[Float], Array[Float])] = runQueriesAgainstDocs(
+      predictionPath,
+      bundlePath,
+      featureConfigPath,
+      "serving_tfrecord_protos",
+      "StatefulPartitionedCall_1"
+    )
+
+    allScores.foreach {
+      case (query: StringMapQueryAndPredictions, sequenceExample: SequenceExample, scores: Array[Float], predictedScores: Array[Float]) =>
+        validateRankingScores(query, sequenceExample, scores, scores.length)
+    }
+  }
+
+  /**
+   * Helper method to produce scores for a model, given input CSV-formatted test data
+   *
+   * @param csvDataPath path to CSV-formatted training/test data
+   * @param modelPath path to SavedModelBundle
+   * @param featureConfigPath path to feature_config.yaml
+   * @param inputTFNode tensorflow graph node name to feed in SequencExamples for scoring
+   * @param scoresTFNode tensorflow graph node name to fetch the scores
+   * @return scores for each input
+   */
+  def runQueriesAgainstDocs(
+                             csvDataPath: String,
+                             modelPath: String,
+                             featureConfigPath: String,
+                             inputTFNode: String,
+                             scoresTFNode: String): Iterable[(StringMapQueryAndPredictions, SequenceExample, Array[Float], Array[Float])] = {
+    val featureConfig = ModelFeaturesConfig.load(featureConfigPath)
+    val sequenceExampleBuilder = StringMapSequenceExampleBuilder.withFeatureProcessors(featureConfig)
+    val rankingModelConfig = ModelExecutorConfig(inputTFNode, scoresTFNode)
+
+    val rankingModel = new TFRecordExecutor(modelPath, rankingModelConfig)
+
+    val queryContextsAndDocs = StringMapCSVLoader.loadDataFromCSV(csvDataPath, featureConfig)
+    assertTrue("attempting to test empty query set!", queryContextsAndDocs.nonEmpty)
+    queryContextsAndDocs.map {
+      case q @ StringMapQueryAndPredictions(queryContext, docs, predictedScores) =>
+        val sequenceExample = sequenceExampleBuilder.build(queryContext.asJava, docs.map(_.asJava).asJava)
+        (q, sequenceExample, rankingModel(sequenceExample), predictedScores)
+    }
+  }
 
   @Test
   def testClassificationGeneratedModelBundle(): Unit = {
     val generatedBundleLocation = System.getProperty("bundleLocation")
-    def modelName = System.getProperty("runName")
+    def modelName = System.getProperty("runName") + "_classification"
     val bundlePath = generatedBundleLocation + "models/" + modelName + "/final/tfrecord"
     val predictionPath = generatedBundleLocation + "logs/" + modelName + "/model_predictions.csv"
     //val featureConfigPath = generatedBundleLocation + "ml4ir/applications/classification/tests/data/configs/feature_config.yaml"
```

Comment:
have we figured out yet why the classification feature_config.yaml doesn't work?  Is it because of the docker image issue again?

Reply:
It doesn't seems to be a docker issue for this one. Maybe a parsing issue with the extra-argument.
I'll add a follow-up work to figure out the problem.